docs/lifetime/index.rst

```diff
@@ -87,6 +87,113 @@ Part of the job of the lifetime scope is to :doc:`handle disposal of the compone
 
+---------------------------
+Example: Singleton Lifetime
+---------------------------
+
+Earlier we noted that a component will get its dependencies from the scope that owns the component. Let's dive into this with an example: singletons.
+
+**When you declare a singleton, it's owned by the scope in which it's registered.**
+
+- If you declare a singleton when building your container, it's held by the root lifetime scope. When you resolve a singleton registered this way, *all of its dependencies will come from the root scope*.
+- When creating a child lifetime scope, you can add singletons - those will be held by the child lifetime scope in which they're registered. When you resolve it, *all of its dependencies will come from that child lifetime scope*.
+
+Doing it this way ensures you can't dispose dependencies out from under the singleton; and that you won't create a memory leak by holding references to things in child scopes after the scope is disposed.
+
+Let's say you have a couple of classes like this:
+
+.. sourcecode:: csharp
+
+    public class Component
+    {
+      private readonly Dependency _dep;
+
+      public string Name => this._dep.Name;
+
+      public Component(Dependency dep)
+      {
+        this._dep = dep;
+      }
+    }
+
+    public class Dependency
+    {
+      public string Name { get; }
+
+      public Dependency(string name)
+      {
+        this.Name = name;
+      }
+    }
+
+Now let's say you have some code that builds a container and uses these classes.
+
+.. sourcecode:: csharp
+
+    var builder = new ContainerBuilder();
+
+    // A singleton declared at container build
+    // time will be owned by the root scope and get
+    // all dependencies from the root scope.
+    builder.RegisterType<Component>()
+           .SingleInstance();
+
+    // Anything that resolves a Dependency from
+    // the root lifetime scope will see the name
+    // as 'root'
+    builder.Register(ctx => new Dependency("root"));
+
+    var container = builder.Build();
+
+    // Resolve the component from the root lifetime scope.
+    var rootComp = container.Resolve<Component>();
+
+    // This will show "root"
+    Console.WriteLine(rootComp.Name);
+
+    // Even if we override the Dependency in a child
+    // scope, it'll still show "root"
+    using (var child1 = container.BeginLifetimeScope(
+      b => b.Register(ctx => new Dependency("child1")))
+    {
+      child1Comp = child1.Resolve<Component>();
+      Console.WriteLine(child1Comp.Name);
+    }
+
+    // You can override the singleton by adding a new
+    // singleton in a child scope. This one will be
+    // owned by the child and will work in its children,
+    // but it doesn't override at the root scope.
+    using (var child2 = container.BeginLifetimeScope(
```

Comment:
Does this actually override the `Component` though? You're only redeclaring the `Dependency` here, but not declaring a new `Component`. Unless I'm missing something obvious.

Reply:
Dammit, I forgot to redeclare. My brain nowadays. Good catch.

Reply:
OK, got it.